### PR TITLE
Fixed extra question mark from being added to initial load of a network link

### PR DESCRIPTION
**Pull request by tfili**
This usually worked except for certain servers with it configured to return a 404 if a ? was added to the url. Not exactly sure how to test this without depending on outside servers.

### CODE REVIEW

**Comment by mramato** [comment]
You can spy on `loadXhr` and make sure it is called with the correct url.  You can also use a fake to have it return whatever data you need to trigger network links to begin with.

**Comment by tfili** [comment]
@mramato I added a bunch of query string related tests. This should be good to go.

**Comment by mramato** [comment]
Update CHANGES

File: Specs/DataSources/KmlDataSourceSpec.js [hunk 1]
```diff
@@ -3022,10 +3026,176 @@ defineSuite([
                 expect(link2).toEqual(link2);
                 expect(folder2).not.toEqual(folder1);
                 expect(placemark2).not.toEqual(placemark1);
+
+                // Reset camera
+                options.camera.positionWC.x = 0.0;
```
Comment (mramato) [inline Specs/DataSources/KmlDataSourceSpec.js]: Should this be in a `beforeEach` instead of here?  Otherwise tests may be order dependent, right?

**Comment by mramato** [comment]
Just those 2 comments then this is good to go.  Thanks @tfili

File: Specs/DataSources/KmlDataSourceSpec.js (hunk 1 above)
Comment (tfili) [inline Specs/DataSources/KmlDataSourceSpec.js]: This is the only test that changes the camera. So I could put it there, but it won't do anything except after this test.

Comment (mramato) [inline Specs/DataSources/KmlDataSourceSpec.js]: Okay for now then.

**Comment by tfili** [comment]
@mramato Updated.

**Pull request merged**